archivematica/startTransfer.py

```diff
@@ -28,8 +25,16 @@
     # start transfer: https://wiki.archivematica.org/Archivematica_API#Start_Transfer
     start = requests.post(full_url, headers=headers, data=params)
     print(start.json()['message'] + time.strftime(" %b %d %H:%M:%S"))
-    time.sleep(30) # pause for 30 seconds
+    time.sleep(5)
     print("Approving {}".format(txfr) + time.strftime(" %b %d %H:%M:%S"))
+    while True:
+        unapproved_transfers = requests.get(os.path.join(baseurl, 'transfer/unapproved'), headers=headers).json().get('results')
+        if unapproved_transfers:
```

Comment:
```suggestion
        if unapproved_transfers and txfr in unapproved_transfers:
```
I *think* this would work, and would ensure that the transfer you try to approve in the next step actually exists, but my assumption is that the name param passed to the `start_transfer` endpoint is the same as the UUID returned by the `unapproved_transfers` endpoint, which may be an erroneous assumption.